### PR TITLE
Handle non default transfer port id

### LINKED ISSUE #396: Handle non default IBC transfer port in message encoder
The IBC transfer module's default port is `transfer`. This can be customized in the genesis though. Our [handler plugin](https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/handler_plugin.go#L260) has the default value hard coded.
Instead the value from genesis should be used.

### PR DESCRIPTION
**Pull request by alpe**
Resolves #396

### CODE REVIEW

File: x/wasm/internal/keeper/keeper.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ func NewKeeper(
 	channelKeeper types.ChannelKeeper,
 	portKeeper types.PortKeeper,
 	capabilityKeeper types.CapabilityKeeper,
+	portSource types.ICS20TransferPortSource,
```
Comment (ethanfrey) [inline x/wasm/internal/keeper/keeper.go:100]: New interfaces for eveything. Yay!

File: x/wasm/internal/keeper/wasmtesting/mock_keepers.go [hunk 2]
```diff
@@ -87,13 +88,24 @@ func (m MockCapabilityKeeper) ClaimCapability(ctx sdk.Context, cap *capabilityty
...
 func (m MockCapabilityKeeper) AuthenticateCapability(ctx sdk.Context, capability *capabilitytypes.Capability, name string) bool {
 	if m.AuthenticateCapabilityFn == nil {
 		panic("not supposed to be called!")
 	}
 	return m.AuthenticateCapabilityFn(ctx, capability, name)
+}
 
+var _ types.ICS20TransferPortSource = &MockIBCTransferKeeper{}
+
+type MockIBCTransferKeeper struct {
```
Comment (ethanfrey) [inline x/wasm/internal/keeper/wasmtesting/mock_keepers.go:102]: Cool to mock

File: x/wasm/internal/keeper/handler_plugin_encoders.go [hunk 3]
```diff
@@ -225,32 +225,33 @@ func EncodeWasmMsg(sender sdk.AccAddress, msg *wasmvmtypes.WasmMsg) ([]sdk.Msg,
...
+				ChannelId: msg.CloseChannel.ChannelID,
+				Signer:    sender.String(),
+			}}, nil
+		case msg.Transfer != nil:
+			amount, err := convertWasmCoinToSdkCoin(msg.Transfer.Amount)
+			if err != nil {
+				return nil, sdkerrors.Wrap(err, "amount")
+			}
+			msg := &ibctransfertypes.MsgTransfer{
+				SourcePort:       portSource.GetPort(ctx),
```
Comment (ethanfrey) [inline x/wasm/internal/keeper/handler_plugin_encoders.go:243]: :100: and this even picks it up if the port changes after genesis

**Review by ethanfrey — APPROVED** [review]
Great stuff

File: x/wasm/internal/keeper/handler_plugin_encoders.go (hunk 3 above)
Comment (alpe) [reply x/wasm/internal/keeper/handler_plugin_encoders.go:243]: ja, theoretically a parameter change proposal may modify it.

**Pull request merged**